dockerfiles/theia/Dockerfile

```diff
@@ -10,33 +10,46 @@
 FROM node:8-alpine
 
 ARG GITHUB_TOKEN
-ARG THEIA_VERSION=0.3.10
-ENV USE_LOCAL_GIT=true \
-    GITHUB_TOKEN=${GITHUB_TOKEN} \
+ARG THEIA_VERSION=next
+
+ENV GITHUB_TOKEN=${GITHUB_TOKEN} \
     THEIA_VERSION=${THEIA_VERSION} \
-    YARN_CACHE_FOLDER=/usr/local/share/.cache/yarn
-
-# build dependencies requireed to compile a custom Theia
-RUN apk add --no-cache make gcc g++ python git openssh bash supervisor jq
-WORKDIR /home/theia
-# build Theia with all extensions to persist yarn cache in the image and
-# have default Theia build in the workspace in case no plugins are requested
-ADD https://raw.githubusercontent.com/theia-ide/theia/v${THEIA_VERSION}/examples/browser/package.json /home/theia/package.json
-ADD theia-default-package.json /home/default/theia/package.json
-ADD versions.sh /home/default/versions.sh
-ADD supervisord.conf /etc/
-RUN cd /home/default && ./versions.sh && \
-    cd /home/theia && \
-    yarn && \
-    yarn theia build --mode development && \
-    rm -rf * && \
-    cd /home/default/theia && \
+    USE_LOCAL_GIT=true \
+    HOME=/home/theia
+
+RUN apk update && apk add --no-cache make gcc g++ python git openssh bash supervisor jq && \
+    rm -rf /tmp/* /var/cache/apk/*
+
+# Packages which will be included into default build of Theia
+ADD theia-default-package.json ${HOME}/package.json
```

Comment:
as we're adding all files into ${HOME}
maybe we could use a single ADD instruction for everything that goes into /home/theia
(instead of one for package.json, one for versions.sh, one for add-extensions.js and one for start.sh

we could have folders like
```
src/home/theia --> /home/theia
src/etc/ ---> /etc/
```

Reply:
To me it is better to keep it as is because that `ADD` instructions appears when target file is needed. For example:
```
ADD src/versions.sh ${HOME}/versions.sh
RUN ${HOME}/versions.sh && rm ${HOME}/versions.sh
```
We add, use and finally remove file. No need to search whole dockerfile and think about context (who else could use or modify or require it).